File: frontend/lib/models/dossier.dart
class Dossier {
  final int id;
  final String numero;
  final int contrat;
  final int adhesion;
  final String nom;
  final String prenom;
  final String? adresse;
  final String? emploi;
  final String malade;
  final String dateNaissance;

  Dossier({
    required this.id,
    required this.numero,
    required this.contrat,
    required this.adhesion,
    required this.nom,
    required this.prenom,
    this.adresse,
    this.emploi,
    required this.malade,
    required this.dateNaissance,
  });

  factory Dossier.fromJson(Map<String, dynamic> json) {
    return Dossier(
      id: json['id'],
      numero: json['numero'],
      contrat: json['contrat'],
      adhesion: json['adhesion'],
      nom: json['nom'],
      prenom: json['prenom'],
      adresse: json['adresse'],
      emploi: json['emploi'],
      malade: json['malade'],
      dateNaissance: json['dateNaissance'],
    );
  }

  String get fullName => '$prenom $nom';
}

File: backend/src/main/java/com/anomalix/backend/model/Dossier.java
package com.anomalix.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "Dossier")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Dossier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true, length = 20)
    private String numero;

    @Column(nullable = false)
    private Integer contrat;

    @Column(nullable = false)
    private Long adhesion;

    @Column(nullable = false, length = 100)
    private String nom;

    @Column(nullable = false, length = 100)
    private String prenom;

    @Column(length = 255)
    private String adresse;

    @Column(length = 100)
    private String emploi;

    @Column(name = "num_adherent", length = 20)
    private String numAdherent;

    @Column(nullable = false, length = 20)
    private String malade; // "adherent", "conjoint", "enfant"

    @Column(name = "date_naissance", nullable = false)
    private LocalDate dateNaissance;
}